pkg/claim/claim.go

```diff
@@ -0,0 +1,140 @@
+package claim
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	ljsonrpc "github.com/lbryio/lbry.go/v2/extras/jsonrpc"
+
+	"go.uber.org/zap"
+)
+
+const (
+	lbrytvAPI = "https://api.lbry.tv/api/v1/proxy"
+	cdnServer = "https://cdn.lbryplayer.xyz/api/v3/streams"
+)
+
+var (
+	lbrytvClient   = ljsonrpc.NewClient(lbrytvAPI)
+	downloadClient = &http.Client{
+		Timeout: 1200 * time.Second,
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout:   15 * time.Second,
+				KeepAlive: 120 * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   30 * time.Second,
+			ResponseHeaderTimeout: 15 * time.Second,
+		},
+	}
+
+	ErrStreamNotFound = errors.New("could not resolve stream URI")
+)
+
+var logger = zap.NewExample().Sugar().Named("video")
+
+type WriteCounter struct {
+	Loaded, Size uint64
+	Started      time.Time
+	URL          string
+}
+
+func (wc *WriteCounter) Write(p []byte) (int, error) {
+	n := len(p)
+	wc.Loaded += uint64(n)
+	progress := float64(wc.Loaded) / float64(wc.Size) * 100.0
+	// if uint64(progress)%5 == 0 {
+	speed := float64(wc.Loaded) / time.Since(wc.Started).Seconds()
+	logger.Debugw("download progress", "url", wc.URL, "size", wc.Size, "percent", fmt.Sprintf("%.2f", progress), "rate", fmt.Sprintf("%.2f", speed))
+	// }
+	return n, nil
+}
+
+type Claim struct {
+	*ljsonrpc.Claim
+	SDHash string
+}
+
+func Resolve(uri string) (*Claim, error) {
```

Comment:
instead of resolving against sdk, i recommend resolving against wallet servers. one less piece of infrastructure to run

i wrote most of the code to do this. needs a bit of robustifying. i'll push my branch and show you

Reply:
It's resolving against lbrytv-api, seemed like the fastest option. Can be easily swapped.